httpbp/server_test.go

```diff
@@ -392,3 +394,72 @@ func TestNewTestBaseplateServer(t *testing.T) {
 		t.Fatalf("Unexpected count value %v", c.count)
 	}
 }
+
+func TestPanicRecovery(t *testing.T) {
+	var pattern httpbp.Pattern = "/test"
+	path := string(pattern)
+	name := "test"
+
+	store := newSecretsStore(t)
+	defer store.Close()
+
+	panicErr := fmt.Errorf("oops")
+
+	bp := baseplate.NewTestBaseplate(baseplate.NewTestBaseplateArgs{
+		Config:          baseplate.Config{Addr: ":8080"},
+		Store:           store,
+		EdgeContextImpl: ecinterface.Mock(),
+	})
+
+	args := httpbp.ServerArgs{
+		Baseplate: bp,
+		Endpoints: map[httpbp.Pattern]httpbp.Endpoint{
+			pattern: {
+				Name:    name,
+				Methods: []string{http.MethodGet},
+				Handle: func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+					panic(panicErr)
+				},
+			},
+		},
+		Middlewares: []httpbp.Middleware{
+			func(name string, next httpbp.HandlerFunc) httpbp.HandlerFunc {
+				return func(ctx context.Context, w http.ResponseWriter, r *http.Request) (err error) {
+					var httpErr httpbp.HTTPError
+					defer func() {
+						if errors.As(err, &httpErr) {
+							if httpErr.Response().Code != http.StatusInternalServerError {
+								t.Errorf(
+									"error code mismatch, expected %d, got %d",
+									http.StatusInternalServerError,
+									httpErr.Response().Code,
+								)
+							}
+							if !errors.Is(httpErr, panicErr) {
+								t.Errorf("expected http error to wrap %v, got %v", panicErr, httpErr.Unwrap())
+							}
+						} else {
+							t.Fatalf("unexpected error type %v", err)
+						}
+					}()
+					return next(ctx, w, r)
```

Comment:
can you wrap the context in one that will allow you to verify that the right keyvals are logged?

Reply:
Do we really have a way to test logging?

Reply:
we actually have, but it's quite involved and not recommended unless you are writing some core logging feature: https://github.com/reddit/baseplate.go/blob/master/log/zap_wrapper_test.go